### PR TITLE
BB2-3506: Added docs regarding revoke token endpoint

### PR DESCRIPTION
**Pull request by stiwarisemanticbits**
**JIRA Ticket:**
[BB2-3506](https://jira.cms.gov/browse/BB2-3506)

### What Does This PR Do?

Added docs regarding revoke token endpoint

### What Should Reviewers Watch For?

See attached image
Screenshot 2024-11-18 at 5 57 56 PM

If you're reviewing this PR, please check for these things in particular:

### What Security Implications Does This PR Have?

Please indicate if this PR does any of the following:

* Adds any new software dependencies
* Modifies any security controls
* Adds new transmission or storage of data
* Any other changes that could possibly affect security?

- [ ] Yes, one or more of the above security implications apply. This PR must not be merged without the ISSO or team security engineer's approval.

### CODE REVIEW

File: _pages/developers.md [hunk 1]
```diff
@@ -438,6 +438,29 @@ curl -X POST "https://sandbox.bluebutton.cms.gov/v2/o/token/" \
     "access_grant_expiration": "2025-09-05 19:17:53Z"
 }
 ~~~
+
+#### Revoking Tokens
```
Comment (jimmyfagan) [inline _pages/developers.md:442]:
```suggestion
#### Revoking tokens
```

Capitalization should match other headers in my opinion, is that okay @jtheil-nava ?

File: _pages/developers.md [hunk 2]
```diff
@@ -438,6 +438,29 @@ curl -X POST "https://sandbox.bluebutton.cms.gov/v2/o/token/" \
...
+
+Developers can revoke an enrollee’s previously granted access. To invalidate the ability to generate new access tokens without authorization, developers can revoke an access token and the underlying data access grant. 
+
+To revoke an access token, POST to the BB2.0 API v2/o/revoke endpoint with the following parameters:
```
Comment (jimmyfagan) [inline _pages/developers.md:446]:
```suggestion
To revoke an access token, POST to the BB2.0 API `/revoke` endpoint with the following parameters:
```

The way the `/token` endpoint is referred to in other sections is more like this. I think we should aim for consistency on that by using `v2/o/revoke` or `/revoke`. Thoughts @jtheil-nava ?

File: _pages/developers.md [hunk 3]
```diff
@@ -438,6 +438,29 @@ curl -X POST "https://sandbox.bluebutton.cms.gov/v2/o/token/" \
...
+* `client_secret`
+* `token`
+* `refresh_token`
+
+##### cURL command
+~~~
+curl -X POST --url 'https://bluebutton.cms.gov/v2/o/revoke/' \
+--header 'content-type: application/x-www-form-urlencoded' \
+-u <client_id>:<client_secret> \
+-d 'token=wDimPGoA8vwXP51kie71vpsy9l17HN'
```
Comment (jimmyfagan) [inline _pages/developers.md:458]:
```suggestion
-d 'token=oQlduHNr09GKCU506GOgp8OarrAy2q'
```

I think we ought to use one of the access tokens from the above examples, the current one is a refresh token. @jtheil-nava does that sound okay?

**Review by jimmyfagan — COMMENTED** [review]
A few minor comments, mostly content related, will ping Jillian so she can chime in.

File: _pages/developers.md (hunk 1 above)
Comment (jtheil-nava) [reply _pages/developers.md:442]: Yes, agreee

File: _pages/developers.md (hunk 2 above)
Comment (jtheil-nava) [reply _pages/developers.md:446]: Agree

File: _pages/developers.md (hunk 3 above)
Comment (jtheil-nava) [reply _pages/developers.md:458]: Yes

**Review by jtheil-nava — APPROVED** [review]
Please see my approval of Jimmy's feedback

**Pull request merged**